src/applications/vaos/redux/api/vaosApi.js

```diff
@@ -0,0 +1,27 @@
+import { createApi } from '@reduxjs/toolkit/query/react';
+import { apiRequestWithUrl } from 'applications/vaos/services/utils';
+
+export const vaosApi = createApi({
+  reducerPath: 'appointmentApi',
+  baseQuery: () => ({ data: null }),
+  endpoints: builder => ({
+    getReferralById: builder.query({
+      async queryFn(referralId) {
+        try {
+          return await apiRequestWithUrl(
+            `/vaos/v2/epsApi/referrals/${referralId}`,
+            {
+              method: 'GET',
+            },
+          );
+        } catch (error) {
+          return {
+            error: { status: error.status || 500, message: error.message },
```

Comment:
I'm a little unfamiliar with how errors should be handled for referrals but as a rule of thumb we generally handle errors elsewhere in appointments using [this utility](https://github.com/department-of-veterans-affairs/vets-website/blob/19bf477347f482d80c9a1b6314460c71bb261183/src/applications/vaos/utils/error.js#L5) which handles a lot of the logging/parsing functions. Is there any reason we don't use that here?

Reply:
Let me look into this. I didn't know about the util function for error handling.